compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -1842,6 +1842,7 @@ final class SearchRoot extends SearchHistory {
           val pruned = prune(List(tree), implicitDictionary.map(_._2).toList, Nil)
           implicitDictionary0 = null
           if (pruned.isEmpty) result
+          else if (pruned.exists(_._2 == EmptyTree)) NoMatchingImplicitsFailure
```

Comment:
Is there a way to provide a more detailed error that would let the user know more about how the search failed?

Reply:
In principle I guess so. But it's not obvious (to me) how to do that.